src/arkode/arkode.c

```diff
@@ -201,28 +199,26 @@ ARKodeMem arkCreate(SUNContext sunctx)
 int arkResize(ARKodeMem ark_mem, N_Vector y0, realtype hscale,
               realtype t0, ARKVecResizeFn resize, void *resize_data)
 {
+  SUNAssignSUNCTX(ark_mem->sunctx);
```

Comment:
I do not understand this as well as I thought.  I'll read back through the developer guide updates, but I don't recall seeing any discussion of why this call is needed at the beginning of each internal routine.

Reply:
I think I get it now.  Let me summarize and you can correct me as needed.

Because the new error checking is done via macros, and those macros internally expect the existence of `SUNCTX` in the current program scope (or equivalently, `sunctx_`), this first line is required so that the expected shortcut variable `sunctx_` has been defined.  Then, all subsequent error-handling macros can reference `sunctx_` as needed, without the solver code (e.g., `arkode.c`) having to pass its context object (e.g., `ark_mem->sunctx`) as an argument to each error-handling macro.

I don't really like this.  While the declared `sunctx_` is local to the function, this feels a lot like when folks use global variables so that they don't have to pass them as function inputs.  At the end of the day, these error-handling macros indeed *depend* on the SUNContext, yet this implementation style hides that dependence from developers, instead requiring a less-intuitive required "call" to `SUNAssignSUNCTX` at the start of every function.  

Can you explain why this approach is preferable to the more straightforward approach wherein error-handling macros expectthe `SUNContext` as an input, and solver codes pass their context object to those macros (e.g., `SUNAssert(SUNMatGetID(A) == SUNMATRIX_BAND, SUN_ERR_ARG_WRONGTYPE, sunctx)`, or `SUNCheckCallLastErrNoRet(N_VPrintFile(ark_mem->Vabstol, outfile, ark_mem->sunctx));`)?

Reply:
I think this is easiest explained over a quick zoom call